arbiter-core/src/agent.rs

```diff
@@ -53,12 +54,8 @@ impl Agent<NotAttached> {
 
     pub fn attach_to_environment(self, environment: &mut crate::environment::Environment) {
         let middleware = RevmMiddleware::new(&self, environment);
-        let agent = Agent::<IsAttached<RevmMiddleware>> {
-            name: self.name,
-            client: Arc::new(middleware),
-            behaviors: self.behaviors,
-        };
-        environment.add_agent(agent);
+        let agent_attached = self.attach_to_client(middleware.into());
```

Comment:
Oh this was smart. I am curious though, why do we need the `.into()`?

Reply:
Otherwise it is
```
expected struct `std::sync::Arc<_>`
   found struct `middleware::RevmMiddleware`
   ```